src/getExecutablePath.cpp

```diff
@@ -134,6 +134,24 @@ std::string getExecutablePath()
     return ec ? "" : p.make_preferred().string();
 }
 
+#elif(BOOST_OS_HAIKU)
+/* #FIXME */
```

Comment:
Probably can port this: https://github.com/rust-lang/rust/blob/83dfe7b27cf2debecebedd3b038f9a1c2e05e051/src/libstd/sys/unix/os.rs#L343

Reply:
Uh, this is cryptic to me.

Reply:
It basically uses `_get_next_image_info` to fill a struct, checks for errors on that call and then accesses `info.name` which I assume is the exe path. I don't know Haiku or `_get_next_image_info` so you may know more. Maybe you can dig deeper. From a quick google search I assume there is a kernel header `image.h` or so which has these functions.

Reply:
Will check, thank you!